filter/filter_test.go

```diff
@@ -215,3 +220,10 @@ func TestFilters(t *testing.T) {
 		})
 	}
 }
+
+func TestBadFilter(t *testing.T) {
+	filter := `s = \xa8*`
+	_, err := compileFilter(filter)
+	assert.Error(t, err, "Received error for bad glob")
+	assert.Contains(t, err.Error(), "non UTF-8 sequence in regexp", "Received good error message for invalid UTF-8 in a regexp")
```

Comment:
Nit: Use assert.ErrorEqual.

Reply:
assert.ErrorEqual() requires the full error message which I think makes the test overly specific (i.e., a minor change to the format of the error message require a test update).  What we really care about is that we've caught the right type of error (invalid UTF-8) and not, say, a zql parse error.